File: typescale.py
def create_type_scale(base, scale, max):
    curr = round(base / scale)
    out = ""
    for i in range(max, 0, -1):
        out += f"\t\t--fs-{i}: {curr}px;\n"
        curr = round(curr * scale)
    return out

def create_media(device, values):
    return f"\t@media ${{device.{device}}} {{\n{create_type_scale(*values)}\t}}\n\n"

# Definiere die Daten in einem Dictionary
size = 6
font_data = {
    "desktopXL":    (16, 1.333,   size),
    "desktop":      (16, 1.25, size),
    "laptop":       (16, 1.2,   size),
    "tablet":       (16, 1.125,  size),
    "tablet_sm":    (16, 1.08,  size),
    "mobile":       (16, 1.067,   size)
}

with open("typescale.js", "w+") as cssfile:
    head = "import { css } from 'styled-components';\nimport { device } from './breakpoints';\n\nexport default css`\n  :root {\n"
    footer = "}\n`;"
    dynamic = ""
    
    for device, values in font_data.items():
        dynamic += create_media(device, values)
    
    cssfile.write(head + dynamic + footer)

File: src/theme/typescale.ts
import { css } from 'styled-components';
import { device } from './breakpoints';

export default css`
  :root {
	@media ${device.desktopXL} {
		--fs-6: 12px;
		--fs-5: 16px;
		--fs-4: 21px;
		--fs-3: 28px;
		--fs-2: 37px;
		--fs-1: 49px;
	}

	@media ${device.desktop} {
		--fs-6: 13px;
		--fs-5: 16px;
		--fs-4: 20px;
		--fs-3: 25px;
		--fs-2: 31px;
		--fs-1: 39px;
	}

	@media ${device.laptop} {
		--fs-6: 13px;
		--fs-5: 16px;
		--fs-4: 19px;
		--fs-3: 23px;
		--fs-2: 28px;
		--fs-1: 34px;
	}

	@media ${device.tablet} {
		--fs-6: 14px;
		--fs-5: 16px;
		--fs-4: 18px;
		--fs-3: 20px;
		--fs-2: 22px;
		--fs-1: 25px;
	}

	@media ${device.tablet_sm} {
		--fs-6: 15px;
		--fs-5: 16px;
		--fs-4: 17px;
		--fs-3: 18px;
		--fs-2: 19px;
		--fs-1: 21px;
	}

	@media ${device.mobile} {
		--fs-6: 15px;
		--fs-5: 16px;
		--fs-4: 17px;
		--fs-3: 18px;
		--fs-2: 19px;
		--fs-1: 20px;
	}

}
`;